src/functions.php

```diff
@@ -75,25 +104,25 @@ function ($error) use (&$exception, &$rejected, &$wait) {
     // argument does not show up in the stack trace in PHP 7+ only.
     $promise = null;
 
+    if ($rejected) {
+        throw $rejectedThrowable;
+    }
+
+    if ($resolved) {
+        return $resolvedValue;
+    }
+
     while ($wait) {
         Loop::run();
     }
 
     if ($rejected) {
-        // promise is rejected with an unexpected value (Promise API v1 or v2 only)
-        if (!$exception instanceof \Throwable) {
-            $exception = new \UnexpectedValueException(
-                'Promise rejected with unexpected value of type ' . (is_object($exception) ? get_class($exception) : gettype($exception))
-            );
-        }
-
-        throw $exception;
+        throw $rejectedThrowable;
```

Comment:
Looks like this line is no longer covered, otherwise LGTM :shipit:

Reply:
Interestingly enough current tests aren't covering that. Looking into either altering or otherwise adding a new test here and to #21

Reply:
@clue Ok added two additional tests ensuring the event loop is started on both a resolving and rejecting pending promises